code/game/objects/items/weapons/soap.dm

```diff
@@ -0,0 +1,159 @@
+//Various Soaps
+
+/obj/item/weapon/soap
+	name = "soap"
+	desc = "A cheap bar of soap. Doesn't smell."
+	gender = PLURAL
+	icon = 'icons/obj/soap.dmi'
+	icon_state = "soap"
+	w_class = 2.0
+	throwforce = 0
+	throw_speed = 4
+	throw_range = 20
+	flags = OPENCONTAINER
+	var/key_data
+
+/obj/item/weapon/soap/nanotrasen
+	desc = "A NanoTrasen-brand bar of soap. Smells of phoron."
+	icon_state = "soapnt"
+
+/obj/item/weapon/soap/plant
+	desc = "A green bar of soap. Smells like dirt and plants."
+
+/obj/item/weapon/soap/deluxe
+	icon_state = "soapdeluxe"
+
+/obj/item/weapon/soap/deluxe/New()

```

Comment:
Use Initialize instead of new

Reply:
Do I just swap out new() with initialise()? That's a legacy thing I didn't make that one.

Reply:
```suggestion
/obj/item/weapon/soap/deluxe/Initialize()
	. = ..()
```

Reply:
Excellent, thanks. I'll test it tomorrow.

Reply:
This is working, thanks! :D